builtin/logical/pki/path_fetch_issuers.go

```diff
@@ -317,6 +357,55 @@ func (b *backend) pathUpdateIssuer(ctx context.Context, req *logical.Request, da
 		modified = true
 	}
 
+	if issuer.AIAURIs == nil && (len(issuerCertificates) > 0 || len(crlDistributionPoints) > 0 || len(ocspServers) > 0) {
+		issuer.AIAURIs = &certutil.URLEntries{}
+	}
+	if issuer.AIAURIs != nil {
+		if len(issuerCertificates) != len(issuer.AIAURIs.IssuingCertificates) {
+			modified = true
+			issuer.AIAURIs.IssuingCertificates = issuerCertificates
+		} else {
+			for index, value := range issuer.AIAURIs.IssuingCertificates {
+				if len(issuerCertificates) < index+1 || issuerCertificates[index] != value {
```

Comment:
While the extra protection I suppose doesn't hurt, since the above if statement has confirmed that the arrays are the same length there isn't a real reason to validate that here as well?

```suggestion
				if issuerCertificates[index] != value {
```

On the other hand we could factor all this out into something that would look like, cut down the repetition and possibly my future copy/paste mistakes that I will make...

```go
if isStringArrayDifferent(issuer.AIAURIs.IssuingCertificates, issuerCertificates) {
    issuer.AIAURIs.IssuingCertificates = issuerCertificates
    modified = true
}

func isStringArrayDifferent(a, b []string) bool {
    if len(a) != len(b) {
        return false
    }
    for i, v := range a {
        if v != b[i] {
            return false
        }
    }
    return true
}
```

Reply:
This is really verbose for what it does - factoring out a utility function would definitely help future readers of the code. Even more so once I realised it's also copied below for the `PATCH` method handler.

Reply:
I've added a type thingy and refactored these into loops. Let me know if anyone has suggestions for changing this or if its good enough for this pass. :-)